### PR TITLE
Disallow `--cgroup-parent` for API clients in "docker run" calls

### PR DESCRIPTION
**Pull request by CpuID**
Disable the use of `CgroupParent` by API clients, only permitted via sockguard flag (for host safety)

Includes passing test coverage, TDD ftw :)

Fixes https://github.com/buildkite/sockguard/issues/13

cc @lox mind giving it a quick once over for sanity?

### CODE REVIEW

File: director_test.go [hunk 1]
```diff
@@ -248,6 +248,15 @@ func TestHandleContainerCreate(t *testing.T) {
 			},
 			esc: 200,
 		},
+		// Defaults + try set a CgroupParent (should fail, only permitted if sockguard started with -cgroup-parent)
+		"containers_create_13": handleCreateTests{
```
Comment (CpuID) [inline director_test.go]: note: I have a WIP feature branch which adds tests as filenames `11` and `12`, and I wasted 30~mins renaming them last time I introduced another test in the middle :) Hence just skipping them as "reserved", and using `13` here, simpler solution.

File: director.go [hunk 2]
```diff
@@ -247,18 +247,22 @@ func (r *rulesDirector) handleContainerCreate(l socketproxy.Logger, req *http.Re
 			return
 		}
 
-		// apply CgroupParent if enabled
+		cgroupParent, ok := decoded["HostConfig"].(map[string]interface{})["CgroupParent"].(string)
+		if ok == false {
+			l.Printf("Denied container create: failed to cast CgroupParent to string")
+			writeError(w, "Denied container create: failed to cast CgroupParent to string", http.StatusUnauthorized)
```
Comment (lox) [inline director.go]: `http.StatusBadRequest` would probably make sense here?

File: director_test.go (hunk 1 above)
Comment (CpuID) [reply director_test.go]: possibly, maybe yea. I haven't used it to date, but open to a PR porting over the fixtures to it :)

one thing I'm considering later: doing tests around `Direct()` to cover off the regex matching also (whats called within each regex path + method match is getting covered here now), and yea it probably makes some sense to look at that as part of expanding the coverage.

File: director.go (hunk 2 above)
Comment (CpuID) [reply director.go]: +1, thx will adjust

**Pull request merged**